lib/web_ui/lib/src/engine/initialization.dart

```diff
@@ -243,22 +244,24 @@ Future<void> initializeEngineUi() async {
 AssetManager get assetManager => _assetManager!;
 AssetManager? _assetManager;
 
-Future<void> _setAssetManager(AssetManager assetManager) async {
+void _setAssetManager(AssetManager assetManager) {
   assert(assetManager != null, 'Cannot set assetManager to null');
   if (assetManager == _assetManager) {
     return;
   }
 
   _assetManager = assetManager;
+}
 
+Future<void> _downloadAssetFonts() async {
   renderer.fontCollection.clear();
 
   if (_assetManager != null) {
-    await renderer.fontCollection.registerFonts(assetManager);
+    await renderer.fontCollection.downloadAssetFonts(_assetManager!);
```

Comment:
The `!` should be unnecessary here. Is the analyzer giving a lint here?

Reply:
it does lint even though its checked that its not null: 

"The argument type 'AssetManager?' can't be assigned to the parameter type 'AssetManager'."

Reply:
I guess it's because this method is async and can be potentially interrupted in the middle of execution so the compiler has no way to know that `_assetManager` wasn't set to `null` in between the null check and this line executing.